vaadin-messages-flow-parent/vaadin-messages-flow/src/main/resources/META-INF/resources/frontend/messageListConnector.js

```diff
@@ -13,25 +13,34 @@
  * License for the specific language governing permissions and limitations under
  * the License.
  */
+
+/**
+ * Maps the given items to a new array of items with formatted time.
+ */
+function formatItems(items, locale) {
+  const formatter = new Intl.DateTimeFormat(locale, {
+    year: 'numeric',
+    month: 'short',
+    day: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric'
+  });
+
+  return items.map((item) =>
+    item.time
+      ? Object.assign(item, {
+          time: formatter.format(new Date(item.time))
+        })
+      : item
+  );
```

Comment:
Considered changing this to:

```js
return items.map((item) => ({
  ...item,
  time: item.time ? formatter.format(new Date(item.time)) : item.time
}));
```

But maintaining the same item references seems quite intentional so didn't want to touch it